From 03cddb8c1dd8912e15b27e5a5c1cb8edcc9350b9 Mon Sep 17 00:00:00 2001
From: Thales Elero Cervi <thaleselero.cervi@windriver.com>
Date: Fri, 7 Jul 2023 09:28:49 -0300
Subject: [PATCH] Update charts requirements to use local server

This change reverts openstack/openstack-helm commit [1] for charts that
the StarlingX OpenStack application is currently building.
That change was removing the helm-toolkit chart dependency from a local
server, since Helm v3 no longer supports "helm serve" [2], and pointing
it to a given openstack-helm-infra directory in which the helm-toolkit
chart should be placed.
The stx-openstack application does not require this change while it is
relying on chartmuseum for serving charts locally [3].
Instead of changing our build instructions and our custom charts,
including charts for other repositories [4], we simply reverts
openstack-helm requirements to use local server again.

[1] c20c1e4400f5935adf0afd0c65bef2bb12af598b
[2] https://helm.sh/docs/topics/v2_v3_migration/
[3] https://opendev.org/starlingx/openstack-armada-app/src/branch/r/
    stx.8.0/stx-openstack-helm-fluxcd/debian/deb_folder/control#L6
[4] https://opendev.org/starlingx/openstack-armada-app/src/branch/r/
    stx.8.0/stx-openstack-helm-fluxcd/debian/meta_data.yaml#L7

Signed-off-by: Thales Elero Cervi <thaleselero.cervi@windriver.com>
Change-Id: Id2ab4adabb21201da229e4242fe06c1ba1bfd463
[ Upversion to epoxy release ]
Signed-off-by: Daniel Caires <DanielMarques.Caires@windriver.com>
---
 aodh/Chart.yaml       | 2 +-
 barbican/Chart.yaml   | 2 +-
 ceilometer/Chart.yaml | 2 +-
 cinder/Chart.yaml     | 2 +-
 glance/Chart.yaml     | 2 +-
 heat/Chart.yaml       | 2 +-
 horizon/Chart.yaml    | 2 +-
 ironic/Chart.yaml     | 2 +-
 keystone/Chart.yaml   | 2 +-
 neutron/Chart.yaml    | 2 +-
 nova/Chart.yaml       | 2 +-
 placement/Chart.yaml  | 2 +-
 12 files changed, 12 insertions(+), 12 deletions(-)

diff --git a/aodh/Chart.yaml b/aodh/Chart.yaml
index a17e1b120..60003eedb 100644
--- a/aodh/Chart.yaml
+++ b/aodh/Chart.yaml
@@ -27,6 +27,6 @@ maintainers:
 
 dependencies:
   - name: helm-toolkit
-    repository: file://../helm-toolkit
+    repository: http://localhost:8879/charts
     version: ">= 0.1.0"
 ...
diff --git a/barbican/Chart.yaml b/barbican/Chart.yaml
index 32fce39e9..8c03673ad 100644
--- a/barbican/Chart.yaml
+++ b/barbican/Chart.yaml
@@ -26,6 +26,6 @@ maintainers:
 
 dependencies:
   - name: helm-toolkit
-    repository: file://../helm-toolkit
+    repository: http://localhost:8879/charts
     version: ">= 0.1.0"
 ...
diff --git a/ceilometer/Chart.yaml b/ceilometer/Chart.yaml
index 1fa2238bc..1e8480879 100644
--- a/ceilometer/Chart.yaml
+++ b/ceilometer/Chart.yaml
@@ -25,6 +25,6 @@ maintainers:
 
 dependencies:
   - name: helm-toolkit
-    repository: file://../helm-toolkit
+    repository: http://localhost:8879/charts
     version: ">= 0.1.0"
 ...
diff --git a/cinder/Chart.yaml b/cinder/Chart.yaml
index e446fed6a..aae0087be 100644
--- a/cinder/Chart.yaml
+++ b/cinder/Chart.yaml
@@ -26,6 +26,6 @@ maintainers:
 
 dependencies:
   - name: helm-toolkit
-    repository: file://../helm-toolkit
+    repository: http://localhost:8879/charts
     version: ">= 0.1.0"
 ...
diff --git a/glance/Chart.yaml b/glance/Chart.yaml
index afbb8bbb2..a1ae6612b 100644
--- a/glance/Chart.yaml
+++ b/glance/Chart.yaml
@@ -26,6 +26,6 @@ maintainers:
 
 dependencies:
   - name: helm-toolkit
-    repository: file://../helm-toolkit
+    repository: http://localhost:8879/charts
     version: ">= 0.1.0"
 ...
diff --git a/heat/Chart.yaml b/heat/Chart.yaml
index dc5f2627f..57e6ad178 100644
--- a/heat/Chart.yaml
+++ b/heat/Chart.yaml
@@ -26,6 +26,6 @@ maintainers:
 
 dependencies:
   - name: helm-toolkit
-    repository: file://../helm-toolkit
+    repository: http://localhost:8879/charts
     version: ">= 0.1.0"
 ...
diff --git a/horizon/Chart.yaml b/horizon/Chart.yaml
index 9e8b96780..6933d6be0 100644
--- a/horizon/Chart.yaml
+++ b/horizon/Chart.yaml
@@ -26,6 +26,6 @@ maintainers:
 
 dependencies:
   - name: helm-toolkit
-    repository: file://../helm-toolkit
+    repository: http://localhost:8879/charts
     version: ">= 0.1.0"
 ...
diff --git a/ironic/Chart.yaml b/ironic/Chart.yaml
index 77eab90d3..d361ba755 100644
--- a/ironic/Chart.yaml
+++ b/ironic/Chart.yaml
@@ -26,6 +26,6 @@ maintainers:
 
 dependencies:
   - name: helm-toolkit
-    repository: file://../helm-toolkit
+    repository: http://localhost:8879/charts
     version: ">= 0.1.0"
 ...
diff --git a/keystone/Chart.yaml b/keystone/Chart.yaml
index 66ccd42bb..62be05939 100644
--- a/keystone/Chart.yaml
+++ b/keystone/Chart.yaml
@@ -26,6 +26,6 @@ maintainers:
 
 dependencies:
   - name: helm-toolkit
-    repository: file://../helm-toolkit
+    repository: http://localhost:8879/charts
     version: ">= 0.1.0"
 ...
diff --git a/neutron/Chart.yaml b/neutron/Chart.yaml
index 01d4ceb4f..31c8c10df 100644
--- a/neutron/Chart.yaml
+++ b/neutron/Chart.yaml
@@ -26,6 +26,6 @@ maintainers:
 
 dependencies:
   - name: helm-toolkit
-    repository: file://../helm-toolkit
+    repository: http://localhost:8879/charts
     version: ">= 0.1.0"
 ...
diff --git a/nova/Chart.yaml b/nova/Chart.yaml
index b421379a3..62c4ba1bc 100644
--- a/nova/Chart.yaml
+++ b/nova/Chart.yaml
@@ -26,6 +26,6 @@ maintainers:
 
 dependencies:
   - name: helm-toolkit
-    repository: file://../helm-toolkit
+    repository: http://localhost:8879/charts
     version: ">= 0.1.0"
 ...
diff --git a/placement/Chart.yaml b/placement/Chart.yaml
index f0f2662a7..be4cc09d6 100644
--- a/placement/Chart.yaml
+++ b/placement/Chart.yaml
@@ -28,6 +28,6 @@ maintainers:
 
 dependencies:
   - name: helm-toolkit
-    repository: file://../helm-toolkit
+    repository: http://localhost:8879/charts
     version: ">= 0.1.0"
 ...
-- 
2.34.1

